Replace e2e timestamp format tests with unit tests (#3894)
bbc/simorgh: src/app/containers/ArticleTimestamp/timeFormatTests/README.md

# Article Timestamp Format tests

> These are unit tests to ensure that we observe and maintain regression for the article timestamp and timezone logic with assiociated timeformats accross services.

Reviewer: ```suggestion
These are unit tests to ensure that we observe and maintain regression for the article timestamp and timezone logic with associated timeformats accross services.
```

> These are unit tests to ensure that we observe and maintain regression for the article timestamp and timezone logic with assiociated timeformats accross services.

Reviewer 2: ```suggestion
These are unit tests to ensure that we observe and maintain regression for the article timestamp and timezone logic with associated time formats across services.
```

> The tests in this directory runs for all services against expectations set in this [file](./expectedFormats.json) and a failure would either indicate a change or breakage in the timestamp or time format logic.

Reviewer: ```suggestion
The tests in this directory runs for all services against expectations set in [expectedFormats.json](./expectedFormats.json) and a failure would either indicate a change or breakage in the timestamp or time format logic.
```

> A change in timestamp logic would be a good reason to update the values in [expectedFormats.json](./expectedFormats.json) but proper investigation should be carried out to identify if there is a valid reason for why these tests fail before running the update script.

Reviewer: ```suggestion
A change in timestamp logic would require updating the values in [expectedFormats.json](./expectedFormats.json) using the [generateTimeFormats.js](./generateTimeFormats.js) script (see below). However, proper investigation should be carried out to identify if there is a valid reason for why these tests fail before running the update script.
```

> A couple of scenarios that we expect these test would fail and need updating would be

Reviewer 2: ```suggestion
A couple of scenarios that we expect these tests would fail and need updating would be:
```

- Change in timestamp or timezone logic in `@bbc/moment-timezone-include`, `@bbc/psammead-locales` etc.

- Change in the timeformat logic [here](../timeFormats) 

- Changes in DST times accross different timezones.

- ... please add more if you find any

## Updating expected time formats
From your local repository run this command from the root directory
```

> npx -p @babel/core -p @babel/node babel-node --presets @babel/preset-env ./src/app/containers/ArticleTimestamp/timeFormatTests/generateTimeFormats.js

Reviewer: This works fine for me without `--presets @babel/preset-env`, is it needed?
